### PR TITLE
update user

### PR DESCRIPTION
**Pull request by rayaviri**
update user

### CODE REVIEW

File: src/main/java/org/jala/efeeder/image/ImageCommand.java [hunk 1]
```diff
@@ -10,6 +10,8 @@
 import java.io.ByteArrayOutputStream;
 import java.io.File;
 import java.io.IOException;
+
```
Comment (ricardo6) [inline src/main/java/org/jala/efeeder/image/ImageCommand.java]: Remove extra space

File: src/main/java/org/jala/efeeder/image/ImageCommand.java [hunk 2]
```diff
@@ -24,21 +26,27 @@
...
+	private static final String USER_IMG_PATH = "/assets/img/empty_user.jpeg";
+	private static final String PLACE_IMG_PATH = "/assets/img/empty_place.jpeg";
+	private static final String FOOD_MEETING_IMG_PATH = "/assets/img/empty_food.jpeg";
+	
+	private String defaultPath = "";
+
+	@Override
+	public Out execute(In parameters) throws Exception {
+		String imageName = parameters.getParameter("file_name");
+
```
Comment (ricardo6) [inline src/main/java/org/jala/efeeder/image/ImageCommand.java:37]: Remove extra spaces

File: src/main/java/org/jala/efeeder/login/LoginCommand.java [hunk 3]
```diff
@@ -1,55 +1,41 @@
...
+ */
+@Command
+public class LoginCommand implements CommandUnit {
+
+	@Override
+	public Out execute(In parameters) throws Exception {
+
+		Out out = new DefaultOut();
+
+		if(parameters.getParameter("username") == null || parameters.getParameter("password") == null)
```
Comment (ricardo6) [inline src/main/java/org/jala/efeeder/login/LoginCommand.java:23]: move up brackets

File: src/main/java/org/jala/efeeder/servlets/CommandServlet.java [hunk 4]
```diff
@@ -1,162 +1,167 @@
...
+import org.jala.efeeder.servlets.support.InBuilder;
+import org.jala.efeeder.user.User;
+
+/**
+ * Created by alejandro on 07-09-16.
+ */
+public class CommandServlet extends HttpServlet {
+
+	private static final long serialVersionUID = 5585317604797123555L;
+
```
Comment (ricardo6) [inline src/main/java/org/jala/efeeder/servlets/CommandServlet.java:39]: Remove extra spaces

File: src/main/java/org/jala/efeeder/user/CreateUserCommand.java [hunk 5]
```diff
@@ -1,57 +1,67 @@
...
+				parameters.getParameter("email"),
+				parameters.getParameter("name"),
+				parameters.getParameter("last_name"),
+				parameters.getParameter("image"),
+				parameters.getParameter("username"),
+				Encrypt.getPasswordEncrypter(parameters.getParameter("password")));
+		try {
+			UserManager userManager = new UserManager(parameters.getConnection());
+
+			if (isNew)
```
Comment (ricardo6) [inline src/main/java/org/jala/efeeder/user/CreateUserCommand.java]: move up the bt

File: src/main/java/org/jala/efeeder/user/UserManager.java [hunk 6]
```diff
@@ -1,64 +1,127 @@
...
+		{
+			user = users.get(0);
+		}
+
+		return user;
+	}
+	
+	public User getUserByUserNamePassword(String userName, String password) throws SQLException {
+		User user = null;
+		if (userName.isEmpty() == false && password.isEmpty() == false) {
```
Comment (ricardo6) [inline src/main/java/org/jala/efeeder/user/UserManager.java:82]: improve conditions

File: src/main/java/org/jala/efeeder/user/UserManager.java [hunk 7]
```diff
@@ -1,64 +1,127 @@
...
+			PreparedStatement preparedStatement = connection.prepareStatement(USERS_BY_USERNAME_PASSWORD_QUERY);
+			preparedStatement.setString(1, userName);
+			preparedStatement.setString(2, password);
+
+			ResultSet resultSet = preparedStatement.executeQuery();
+
+			List<User> users = getUsersFromResultSet(resultSet);
+			if (users.isEmpty() == false)
```
Comment (ricardo6) [inline src/main/java/org/jala/efeeder/user/UserManager.java]: improve

File: src/main/webapp/WEB-INF/user/user.jsp [hunk 8]
```diff
@@ -1,75 +1,84 @@
...
+<%@ taglib prefix="t" tagdir="/WEB-INF/tags" %>
+
+
+<t:template>
+	<jsp:attribute name="javascript">
+		<script src="/assets/js/user.js"></script>
+	</jsp:attribute>
+	
+	<jsp:body>
+		<div class="row" style="margin-top: 10%;">
```
Comment (ricardo6) [inline src/main/webapp/WEB-INF/user/user.jsp]: move style to a css file

File: src/main/webapp/WEB-INF/user/user.jsp [hunk 9]
```diff
@@ -1,75 +1,84 @@
...
+				<div class="col offset-m3 s12 m6">
+					<div class="card">
+						<div class="card-content">
+							<form id="create-user-form" role="form" action="CreateUser" method="post" enctype="multipart/form-data">
+								<div class="row center-align">
+									<div id="image-preview" class="circle responsive-img">
+										<label for="image-upload" id="image-label" class="center">
+											<image class="large material-icons circle responsive-img" src="image?file_name=${user.getImage()}&type=user"/>
+										</label>
+										<input type="file" name="image" id="image-upload" style="display:none;"/>
```
Comment (ricardo6) [inline src/main/webapp/WEB-INF/user/user.jsp:29]: move style to a css file

File: src/main/webapp/WEB-INF/user/user.jsp [hunk 10]
```diff
@@ -1,75 +1,84 @@
...
+										<label for="confirm_password" >Confirm Password.</label>
+									</div>
+								</div>
+								<div class="row">
+									<div class="right">
+										<input type="hidden" id="new-user-hiden" name="inputName" value="${newUser}">
+										<button id="cancel-button" class="btn btn-primary" type="button">Cancel</button>
+										<c:choose>
+											<c:when test="${newUser}">
+												<button id="create-button" class="btn btn-primary" type="submit">Add User</button>
```
Comment (ricardo6) [inline src/main/webapp/WEB-INF/user/user.jsp:67]: try not use breaks, instead use css to move elements

File: src/main/java/org/jala/efeeder/image/ImageCommand.java [hunk 11]
```diff
@@ -47,31 +55,39 @@ public Out execute(In parameters) throws Exception {
...
+	
+	private Out readImageAndConvertToBytes(String path, String formatName, String contentTypeName) throws IOException{
+		File imagenFile = new File(path);
+		ByteArrayOutputStream baos = new ByteArrayOutputStream(1000);
+		BufferedImage img = null;
 
-        String base64String = Base64.encode(baos.toByteArray());
-        baos.close();
+		try{
+
```
Comment (ziur) [inline src/main/java/org/jala/efeeder/image/ImageCommand.java]: Remove line

File: src/main/java/org/jala/efeeder/image/ImageCommand.java [hunk 12]
```diff
@@ -47,31 +54,38 @@ public Out execute(In parameters) throws Exception {
...
-        File imagenFile = new File(path);
-        ByteArrayOutputStream baos = new ByteArrayOutputStream(1000);
-        BufferedImage img = ImageIO.read(imagenFile);
-        
-        ImageIO.write(img, formatName, baos);
-        baos.flush();
+	}
+	
+	private Out readImageAndConvertToBytes(String path, String formatName, String contentTypeName) throws IOException{
+		File imagenFile = new File(path);
```
Comment (MirkoTV) [inline src/main/java/org/jala/efeeder/image/ImageCommand.java]: Please change this "imagenFile" name

File: src/main/java/org/jala/efeeder/login/LoginCommand.java (hunk 3 above)
Comment (MirkoTV) [reply src/main/java/org/jala/efeeder/login/LoginCommand.java:23]: yes, lets follow java conventions for curly braces.

File: src/main/java/org/jala/efeeder/user/UpdateUserCommand.java [hunk 13]
```diff
@@ -0,0 +1,29 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package org.jala.efeeder.user;
```
Comment (MirkoTV) [inline src/main/java/org/jala/efeeder/user/UpdateUserCommand.java]: Remove this header please

File: src/main/java/org/jala/efeeder/user/User.java [hunk 14]
```diff
@@ -1,53 +1,61 @@
...
-	{
-		return this.name + " " + this.lastName;
-	}
-}
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package org.jala.efeeder.user;
```
Comment (MirkoTV) [inline src/main/java/org/jala/efeeder/user/User.java:1]: Lets remove this header please

File: src/main/java/org/jala/efeeder/user/UserCommand.java [hunk 15]
```diff
@@ -1,26 +1,29 @@
...
-
-        return out.forward("user/user.jsp");
-    }
-}
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package org.jala.efeeder.user;
```
Comment (MirkoTV) [inline src/main/java/org/jala/efeeder/user/UserCommand.java:1]: Lets remove this header please

File: src/main/java/org/jala/efeeder/user/UserManager.java (hunk 6 above)
Comment (MirkoTV) [reply src/main/java/org/jala/efeeder/user/UserManager.java:82]: @ricardo6 how do you think we can improve this conditions?

File: src/main/java/org/jala/efeeder/user/UserManager.java [hunk 16]
```diff
@@ -1,64 +1,127 @@
...
+
+	public User getUserById(int id) throws SQLException {
+
+		User user = null;
+		List<Integer> listId = new ArrayList<>();
+		listId.add(id);
+		List<User> users = getUsersById(listId);
+		if (users.isEmpty() == false)
+		{
+			user = users.get(0);
```
Comment (MirkoTV) [inline src/main/java/org/jala/efeeder/user/UserManager.java:74]: follo java conventions for curly braces please

File: src/main/java/org/jala/efeeder/user/UserManager.java [hunk 17]
```diff
@@ -1,64 +1,127 @@
...
+		if (userName.isEmpty() == false && password.isEmpty() == false) {
+			PreparedStatement preparedStatement = connection.prepareStatement(USERS_BY_USERNAME_PASSWORD_QUERY);
+			preparedStatement.setString(1, userName);
+			preparedStatement.setString(2, password);
+
+			ResultSet resultSet = preparedStatement.executeQuery();
+
+			List<User> users = getUsersFromResultSet(resultSet);
+			if (users.isEmpty() == false)
+			{
```
Comment (MirkoTV) [inline src/main/java/org/jala/efeeder/user/UserManager.java]: move up curly braces

File: src/main/webapp/assets/js/user.js [hunk 18]
```diff
@@ -1,132 +1,139 @@
...
-		init: function(){
-			addValidateRules();
-			addEventClick();
-		},
-	};
-};
-
-
-
+$(document).ready(function () {
```
Comment (MirkoTV) [inline src/main/webapp/assets/js/user.js]: Lets use the jquery shorthand for document ready

File: src/main/webapp/assets/js/user.js [hunk 19]
```diff
@@ -1,132 +1,139 @@
...
+		});
+	};
+
+	return {
+		init: function(){
+			addValidateRules();
+			addEventClick();
+		},
+	};
+};
```
Comment (MirkoTV) [inline src/main/webapp/assets/js/user.js:139]: This file looks better now!

**Review by MirkoTV — CHANGES_REQUESTED** [review]

File: src/main/java/org/jala/efeeder/image/ImageCommand.java (hunk 12 above)
Comment (rayaviri) [reply src/main/java/org/jala/efeeder/image/ImageCommand.java]: este no es mi cambio, solo puse los tabs

**Review by MirkoTV — APPROVED** [review]

**Pull request merged**